src/main/java/chess/domain/Board.java

```diff
@@ -0,0 +1,60 @@
+package chess.domain;
+
+import chess.domain.piece.Color;
+import chess.domain.piece.Piece;
+import chess.domain.piece.moveRule.*;
+import chess.domain.position.File;
+import chess.domain.position.Position;
+import chess.domain.position.Rank;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Board {
+    private final Map<Position, Piece> board;
+
+    public Board() {
+        board = new HashMap<>();
+        initializePiece();
+    }
+
+    private void initializePiece() {
+        initializePawnLinePieces(Rank.RANK2, Color.WHITE);
+        initializePawnLinePieces(Rank.RANK7, Color.BLACK);
+
+        initializeEndLinePieces(Rank.RANK1, Color.WHITE);
+        initializeEndLinePieces(Rank.RANK8, Color.BLACK);
+    }
+
+    private void initializePawnLinePieces(Rank rank, Color color) {
+        for (File file : File.values()) {
+            board.put(Position.of(file, rank), new Piece(PawnMoveRule.of(color), color));
+        }
+    }
+
+    private void initializeEndLinePieces(Rank rank, Color color) {
+        board.put(Position.of(File.FILE_A, rank), new Piece(RookMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_B, rank), new Piece(KnightMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_C, rank), new Piece(BishopMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_D, rank), new Piece(QueenMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_E, rank), new Piece(KingMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_F, rank), new Piece(BishopMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_G, rank), new Piece(KnightMoveRule.getInstance(), color));
+        board.put(Position.of(File.FILE_H, rank), new Piece(RookMoveRule.getInstance(), color));
+    }
+
+    public Map<Position, String> move(Position currentPosition, Position nextPosition) {
+        Piece piece = board.get(currentPosition);
+        piece.move(currentPosition, nextPosition, this.board);
+        return getPrintingBoard();
+    }
+
+    public Map<Position, String> getPrintingBoard() {
+        Map<Position, String> pieceNames = new HashMap<>();
+        for (Position position : board.keySet()) {
+            Piece piece = board.get(position);
+            pieceNames.put(position, piece.formatName());
+        }
+        return pieceNames;
+    }
```

Comment:
해당 [문서](https://rules.sonarsource.com/java/RSPEC-2864)를 참고해서 로직을 수정해보면 어떨까요?
```suggestion
    public Map<Position, String> getPrintingBoard() {
        Map<Position, String> pieceNames = new HashMap<>();
        for (Map.Entry<Position, Piece> entry : board.entrySet()) {
            Piece piece = entry.getValue();
            pieceNames.put(entry.getKey(), piece.formatName());
        }
        return pieceNames;
    }
```

Reply:
어차피 key와 value가 다 필요하다면 entrySet()을 이용하는 게
효율적이겠군요! 이런 세세한 부분까지 봐주셔서 감사합니다 😊
수정하겠습니다.